gpu_support/add_gpu_support.sh

```diff
@@ -0,0 +1,192 @@
+else
+  echo "nvidia-smi not found, exiting now..." >&2
+  exit 1
+fi
+
+# set up basic environment variables, EasyBuild and Lmod
+EESSI_SILENT=1 source /cvmfs/pilot.eessi-hpc.org/versions/2021.12/init/bash
+
+current_dir=$(dirname $(realpath $0))
+
+# Get arch type from EESSI environment
+eessi_cpu_family="${EESSI_CPU_FAMILY:-x86_64}"
+
+# Get OS family
+# TODO: needs more thorough testing
+os_family=$(uname | tr '[:upper:]' '[:lower:]')
+
+# Get OS version
+# TODO: needs more thorough testing, taken from https://unix.stackexchange.com/a/6348
+if [ -f /etc/os-release ]; then
+  # freedesktop.org and systemd
+  . /etc/os-release
+  os=$NAME
+  ver=$VERSION_ID
+  if [[ "$os" == *"Rocky"* ]]; then
+    os="rhel"
+  fi
+  if [[ "$os" == *"Debian"* ]]; then
+    os="debian"
+  fi
+elif type lsb_release >/dev/null 2>&1; then
+  # linuxbase.org
+  os=$(lsb_release -si)
+  ver=$(lsb_release -sr)
+elif [ -f /etc/lsb-release ]; then
+  # For some versions of Debian/Ubuntu without lsb_release command
+  . /etc/lsb-release
+  os=$DISTRIB_ID
+  ver=$DISTRIB_RELEASE
+elif [ -f /etc/debian_version ]; then
+  # Older Debian/Ubuntu/etc.
+  os=Debian
+  ver=$(cat /etc/debian_version)
+else
+  # Fall back to uname, e.g. "Linux <version>", also works for BSD, etc.
+  os=$(uname -s)
+  ver=$(uname -r)
+fi
+# Convert OS version to major versions, e.g. rhel8.5 -> rhel8
+# TODO: needs testing for e.g. Ubuntu 20.04
+ver=${ver%.*}
+
+##############################################################################################
+# Check that the CUDA driver version is adequate
+# (
+#  needs to be r450 or r470 which are LTS, other production branches are acceptable but not
+#  recommended, below r450 is not compatible [with an exception we will not explore,see
+#  https://docs.nvidia.com/datacenter/tesla/drivers/#cuda-drivers]
+# )
+# only check first number in case of multiple GPUs
+driver_version=$(nvidia-smi --query-gpu=driver_version --format=csv,noheader | tail -n1)
+driver_version="${driver_version%%.*}"
+# Now check driver_version for compatability
+# Check driver is at least LTS driver R450, see https://docs.nvidia.com/datacenter/tesla/drivers/#cuda-drivers
+if (( $driver_version < 450 )); then
+  echo "Your NVIDIA driver version is too old, please update first.."
+  exit 1
+fi
+
+
+# Check if the CUDA compat libraries are installed and compatible with the target CUDA version
+# if not find the latest version of the compatibility libraries and install them
+
+# get URL to latest CUDA compat libs, exit if URL is invalid
+latest_cuda_compat_url="$($(dirname "$BASH_SOURCE")/get_latest_cuda_compatlibs.sh ${os} ${ver} ${eessi_cpu_family})"
+ret=$?
+if [ $ret -ne 0 ]; then
+  echo $latest_cuda_compat_url
+  exit 1
+fi
+
+# Create a general space for our NVIDIA compat drivers
+if [ -w /cvmfs/pilot.eessi-hpc.org/host_injections ]; then
+  mkdir -p /cvmfs/pilot.eessi-hpc.org/host_injections/nvidia
+  cd /cvmfs/pilot.eessi-hpc.org/host_injections/nvidia
```

Comment:
Before dropping in here, I would store where we started from and return us there at the end of the script (or at any earlier exit point)

Reply:
Ah, sorry, you have that